js/runner.go

```diff
@@ -122,6 +124,42 @@ func (r *Runner) NewVU(samplesOut chan<- stats.SampleContainer) (lib.VU, error)
 	return lib.VU(vu), nil
 }
 
+func getRandomIP(ifacesList string) (*net.IPAddr, error) {
+	var addresses []net.Addr
+
+	for _, ifaceName := range strings.Split(ifacesList, ",") {
+		targetIf, err := net.InterfaceByName(ifaceName)
+		if err != nil {
+			return nil, errors.New("Cannot find network interface '" + ifaceName + "'")
+		}
+
+		addrs, err := targetIf.Addrs()
+		if err != nil {
+			return nil, errors.New("Cannot get addresses on network interface '" + ifaceName + "'")
+		}
+
+		addresses = append(addresses, addrs...)
+	}
+
+	if len(addresses) == 0 {
+		return nil, errors.New("Cannot find any usable addresses on network interface(s) '" + ifacesList + "'")
+	}
+
+	rand.Shuffle(len(addresses), func(i, j int) { addresses[i], addresses[j] = addresses[j], addresses[i] })
```

Comment:
It doesn't make sense to shuffle an array every time you want to choose a random member of it... You can either use something like `addr := addresses[rand.Int()]`, or better yet, also do the array shuffle only once in the `sync.Once()` block I mentioned above. Then you can just iterate through it (with an atomic counter to avoid races) like this:
```go
myIPIndex := atomic.AddUint32(&globalIPAddrCounter, 1) -1
myIPAddr := addresses[myIPIndex % len(addresses)]
```
Also, if you're using random numbers of shuffling, you should probably make sure that you `Seed()` the random number generator, or better yet, make your own local one with https://golang.org/pkg/math/rand/#NewSource